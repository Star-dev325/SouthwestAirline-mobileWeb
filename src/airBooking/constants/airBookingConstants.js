export const ANALYTICS = {
  FLIGHT_CONFIRMATION_PAGE: {
    page: 'air-booking-confirmation',
    page_channel: 'BOOK',
    page_name: 'confirmation',
    page_subchannel: 'SWA'
  },
  frequentTraveler: {
    page_channel: "BOOK",
    page_name: 'frequent traveler',
    page_subchannel: "SWA"
  },
  MULTI_SELECT_INBOUND_FARE_PAGE: {
    page: 'air-booking-select-multi-inbound-fare',
    selectpageindex: '1'
  },
  MULTI_SELECT_INBOUND_PAGE: {
    page: 'air-booking-select-multi-inbound',
    selectpageindex: '1'
  },
  MULTI_SELECT_OUTBOUND_BOUNDS_PAGE: {
    page: 'air-booking-select-multi-outbound',
    selectpageindex: '0'
  },
  MULTI_SELECT_OUTBOUND_FARE_PAGE: {
    page: 'air-booking-select-multi-outbound-fare',
    selectpageindex: '0'
  },
  MULTI_SELECT_OUTBOUND_PAGE: {
    page: 'air-booking-select-multi',
    selectpageindex: '0'
  },
  MULTI_SELECT_PAGE_CONSTANTS: {
    page_channel: 'air',
    page_name: 'shopping',
    page_subchannel: 'booking'
  },
  PRICE_PAGE: {
    page: 'air-booking-pricing-summary',
    page_channel: 'BOOK',
    page_name: 'Price Page',
    page_subchannel: 'SWA'
  },
  PURCHASE_PAGE_LOAD: 'air booking purchase page',
  YOUNG_TRAVELER_PAGE: {
    page: 'air-booking-young-traveler',
    page_channel: 'BOOK',
    page_name: 'young-traveler',
    page_subchannel: 'SWA'
  }
};
export const CHASE = {
  PAYMENT_FAILURE_ERROR_CODE: 404519017
};
export const CHASE_AUTO_PROVISIONING = {
  CHASE_CARD_PROVISION: 'CHASE_CARD_PROVISION',
  CHASE_EMAIL_PROVISION: 'CHASE_EMAIL_PROVISION'
};
export const TRAVEL_FUNDS = {
  TOKEN_EXPIRED_AIRBOOKING_CODE: 400310756
};
export const SPLIT_PAY_CALC_FUNDS_HREF = '/v1/mobile-air-booking/page/x-calculate-funds/SPLIT_PAYMENT';
