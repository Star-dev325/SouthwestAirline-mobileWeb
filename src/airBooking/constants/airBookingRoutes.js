import { carBookingOldRoutes, carBookingRoutes } from 'src/carBooking/constants/carBookingRoutes';
import { earlyBirdOldRoutes, earlyBirdRoutes } from 'src/earlyBird/constants/earlyBirdRoutes';

export const airBookingRoutes = {
  addManualIrn: '/air/booking/manual-irn',
  applyRapidRewards: '/air/booking/apply-rapid-rewards?clearFormData=false',
  applyTravelFunds: '/air/booking/apply-travel-funds',
  billingAddress: '/air/booking/billing-address',
  carBookingIndex: carBookingRoutes['index'],
  checkin: earlyBirdRoutes['checkin'],
  confirmation: '/air/booking/confirmation.html',
  contactInfoTravelManager: '/air/booking/contact-info-travel-manager',
  contactMethod: '/air/booking/contact-method',
  fareDetails: '/air/booking/fare-details',
  flightShoppingDepart: '/air/booking/select-depart.html',
  flightShoppingReturn: '/air/booking/select-return.html',
  flightShoppingOutbound: '/air/booking/outbound/results',
  frequentTravelers: '/air/booking/passenger/:paxNumber/frequent-travelers',
  index: {
    canonicalPath: '/air/booking/',
    htmlPath: '/air/booking/index.html'
  },
  indexWithoutClearForm: '/air/booking?clearFormData=false',
  internalReferenceNumber: '/air/booking/irn-info',
  lowFareCalendar: {
    canonicalPath: '/air/low-fare-calendar/',
    htmlPath: '/air/low-fare-calendar/select-dates.html'
  },
  lowFareCalendarDate: '/air/low-fare-calendar/date-select',
  passengers: '/air/booking/passenger',
  passengersWithPassport: '/air/booking/passenger/:paxNumber/passport',
  passengersWithPax: '/air/booking/passenger/:paxNumber',
  passengersWithPaxEdit: '/air/booking/passenger/:paxNumber/edit',
  passengersWithSpecialAssistance: '/air/booking/passenger/:paxNumber/special-assistance',
  payment: '/air/booking/payment',
  paymentEdit: '/air/booking/payment/edit',
  price: '/air/booking/price.html',
  priceReview: '/air/booking/price/detail',
  purchase: '/air/booking/purchase.html',
  purchasePaypal: '/air/booking/purchase/paypal',
  purchasePaypalCanceled: '/air/booking/purchase/paypal-canceled',
  purchaseWithoutClearForm: '/air/booking/purchase.html?clearFormData=false',
  recent: '/air/booking/recent',
  reprice: '/air/booking/reprice',
  selectCompany: '/air/booking/select-company',
  selectCompanyToggle: '/air/booking/select-company?CLK=swabizomnitoggle',
  selectDepartFare: '/air/booking/select-fare-depart.html',
  selectFare: '/air/booking/:direction/select-fare',
  selectPassengers: '/air/booking/select-passenger',
  selectReturnFare: '/air/booking/select-fare-return.html',
  viewReservationView: '/air/manage-reservation/view/:recordLocator',
  youngTraveler: '/air/booking/young-traveler',
  youngTravelerEditWithoutClearForm: '/air/booking/young-traveler/edit?clearFormData=false',
  youngTravelerParentConsent: '/air/booking/young-traveler-parent-consent'
};

export const airBookingOldRoutes = {
  addManualIrn: '/air/booking/addManualIrn',
  applyRapidRewards: '/air/booking/apply-rapid-rewards?clearFormData=false',
  carBookingIndex: carBookingOldRoutes['index'],
  checkin: earlyBirdOldRoutes['checkin'],
  confirmation: '/air/booking/confirmation',
  fareDetails: '/air/booking/shopping/fare-details',
  flightShoppingDepart: '/air/booking/shopping/:paxType/:direction/results',
  flightShoppingReturn: '/air/booking/shopping/:paxType/:direction/results',
  flightShoppingOutbound: '/air/booking/shopping/adult/outbound/results',
  frequentTravelers: '/air/booking/passengers/:paxNumber/frequent-travelers',
  index: '/air/booking/shopping',
  indexWithoutClearForm: '/air/booking/shopping?clearFormData=false',
  internalReferenceNumber: '/air/booking/irnInfo',
  lowFareCalendar: '/air/booking/shopping/low-fare-calendar',
  lowFareCalendarDate: '/air/booking/shopping/low-fare-calendar/date-select',
  passengers: '/air/booking/passengers',
  passengersWithPassport: '/air/booking/passengers/:paxNumber/passport',
  passengersWithPax: '/air/booking/passenger/:paxNumbers',
  passengersWithPaxEdit: '/air/booking/passengers/:paxNumber/edit',
  passengersWithSpecialAssistance: '/air/booking/passengers/:paxNumber/special-assistance',
  price: '/air/booking/pricing/summary',
  priceReview: '/air/booking/pricing/review',
  purchase: '/air/booking/review',
  purchaseWithoutClearForm: '/air/booking/review?clearFormData=false',
  recent: '/air/booking/shopping/recent',
  reprice: '/air/booking/pricing/repricing',
  selectCompany: '/air/booking/shopping/select-company',
  selectCompanyToggle: '/air/booking/shopping/select-company?CLK=swabizomnitoggle',
  selectDepartFare: '/air/booking/shopping/:paxType/:direction/select-fare',
  selectFare: '/air/booking/shopping/:paxType/:direction/select-fare',
  selectPassengers: '/air/booking/shopping/select-passengers',
  selectReturnFare: '/air/booking/shopping/:paxType/:direction/select-fare',
  viewReservationView: '/view-reservation/trip-details/:recordLocator',
  youngTraveler: '/air/booking/young-traveler',
  youngTravelerEditWithoutClearForm: '/air/booking/young-traveler/edit?clearFormData=false',
  youngTravelerParentConsent: '/air/booking/young-traveler-parent-consent'
};
