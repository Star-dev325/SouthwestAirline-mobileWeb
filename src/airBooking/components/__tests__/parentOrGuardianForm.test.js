import { render } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import ParentOrGuardianForm from 'src/airBooking/components/parentOrGuardianForm';
import { AIR_BOOKING_PARENT_OR_GUARDIAN_FORM } from 'src/shared/constants/formIds';
import { getPassengerValidationDetails } from 'test/builders/model/youngTravelerPageBuilder';
import { createMockedFormStore } from 'test/unit/helpers/formTestUtils';

describe('ParentOrGuardianForm', () => {
  it('should render the component correctly', () => {
    const { container } = createComponent();

    expect(container).toMatchSnapshot();
  });

  it('should render the component correctly when isEditMode is true', () => {
    const { container } = createComponent({ isEditMode: true });

    expect(container).toMatchSnapshot();
  });

  const createComponent = (props = {}, state = {}) => {
    const { body, disclaimerText, linkText } =
      getPassengerValidationDetails().passengerValidationDetails.youngTraveler.youngTravelerPageInfo;
    const defaultProps = {
      disclaimerText,
      formId: AIR_BOOKING_PARENT_OR_GUARDIAN_FORM,
      infoText: body,
      linkText,
      onClickYoungTravelerParentConsent: () => {},
      onSubmit: () => {}
    };
    const finalProps = {
      ...defaultProps,
      ...props
    };

    return render(
      <Provider store={createMockedFormStore(state)}>
        <ParentOrGuardianForm {...finalProps} />
      </Provider>
    );
  };
});
