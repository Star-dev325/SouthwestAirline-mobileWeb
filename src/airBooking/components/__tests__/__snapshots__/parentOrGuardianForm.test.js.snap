// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParentOrGuardianForm should render the component correctly 1`] = `
<div>
  <form
    class="form"
    name="AIR_BOOKING_PARENT_OR_GUARDIAN_FORM"
    novalidate=""
  >
    <fieldset>
      <div
        class="parent-or-guardian-form--container"
      >
        <p
          class="parent-or-guardian-form--info-text"
        >
          You must complete the form entirely including a phone number your parent/guardian contact can be reached in case of emergency. Once you have completed this purchase, any changes to the parent/guardian information must be completed at the airport.
        </p>
        <div
          class="fields pt5 grouped"
        >
          <label
            class="fields--label"
          >
            AIR_BOOKING__PARENT_OR_GUARDIAN_FORM__PARENT_OR_GUARDIAN_LABEL
          </label>
          <div
            class="field"
          >
            <div
              class="input"
            >
              <input
                class=""
                maxlength="30"
                name="firstName"
                placeholder="SHARED__PLACEHOLDER__FIRST_NAME"
                value=""
              />
            </div>
          </div>
          <div
            class="field"
          >
            <div
              class="input"
            >
              <input
                class=""
                maxlength="30"
                name="lastName"
                placeholder="SHARED__PLACEHOLDER__LAST_NAME"
                value=""
              />
            </div>
          </div>
          <div
            class="field"
          >
            <div
              class="input"
            >
              <input
                class=""
                maxlength="20"
                name="relationship"
                placeholder="SHARED__PLACEHOLDER__RELATIONSHIP"
                value=""
              />
            </div>
          </div>
        </div>
        <div>
          <div
            class="fields mt5 grouped"
          >
            <label
              class="fields--label"
            >
              AIR_BOOKING__PARENT_OR_GUARDIAN_FORM__CONTACT_INFORMATION_LABEL
            </label>
            <div
              class="field"
            >
              <div
                class="country-code-nav-item-field"
                data-qa="country-code-nav-item-field"
              >
                <a
                  class="nav-item-link input"
                >
                  <div
                    class="country-code-nav-item-field--value"
                    data-qa="country-code-nav-item-field-value"
                  >
                    United States of America
                     - 
                    US
                  </div>
                  <i
                    class="nav-item-link--icon icon icon_keyboard-arrow-right"
                  />
                </a>
              </div>
            </div>
            <div
              class="field"
            >
              <div
                class="input large"
              >
                <input
                  class=""
                  maxlength="40"
                  name="addressLine1"
                  placeholder="Street address"
                  value=""
                />
              </div>
            </div>
            <div
              class="field"
            >
              <div
                class="input large"
              >
                <input
                  class=""
                  maxlength="40"
                  name="addressLine2"
                  placeholder="Street address 2 (optional)"
                  value=""
                />
              </div>
            </div>
            <div
              class="field"
            >
              <div
                class="input large"
              >
                <input
                  name="zipOrPostalCode"
                  placeholder="ZIP code"
                  type="tel"
                  value=""
                />
              </div>
            </div>
            <div
              class="field"
            >
              <div
                class="input large"
              >
                <input
                  class=""
                  name="city"
                  placeholder="City"
                  value=""
                />
              </div>
            </div>
            <div
              class="field"
            >
              <div
                class="province-field_us select-input icon-fixed formatted input large"
              >
                <span
                  class="formatted-input formatted-input--fill"
                >
                  <select
                    class="dropdown empty large"
                    data-qa="select-dropdown"
                    name="stateProvinceRegion"
                    placeholder="State"
                  >
                    <option
                      class="placeholder"
                      data-qa="dqa-enabled-option"
                      value=""
                    >
                      State
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="AL"
                    >
                      Alabama
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="AK"
                    >
                      Alaska
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="AS"
                    >
                      American Samoa
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="AZ"
                    >
                      Arizona
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="AR"
                    >
                      Arkansas
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="AA"
                    >
                      Armed Forces Americas (except Canada)
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="AE"
                    >
                      Armed Forces
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="AP"
                    >
                      Armed Forces Pacific
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="CA"
                    >
                      California
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="CO"
                    >
                      Colorado
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="CT"
                    >
                      Connecticut
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="DE"
                    >
                      Delaware
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="DC"
                    >
                      District of Columbia
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="FM"
                    >
                      Federated States of Micronesia
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="FL"
                    >
                      Florida
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="GA"
                    >
                      Georgia
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="GU"
                    >
                      Guam
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="HI"
                    >
                      Hawaii
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="ID"
                    >
                      Idaho
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="IL"
                    >
                      Illinois
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="IN"
                    >
                      Indiana
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="IA"
                    >
                      Iowa
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="KS"
                    >
                      Kansas
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="KY"
                    >
                      Kentucky
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="LA"
                    >
                      Louisiana
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="ME"
                    >
                      Maine
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="MH"
                    >
                      Marshall Islands
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="MD"
                    >
                      Maryland
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="MA"
                    >
                      Massachusetts
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="MI"
                    >
                      Michigan
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="MN"
                    >
                      Minnesota
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="MS"
                    >
                      Mississippi
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="MO"
                    >
                      Missouri
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="MT"
                    >
                      Montana
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="NE"
                    >
                      Nebraska
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="NV"
                    >
                      Nevada
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="NH"
                    >
                      New Hampshire
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="NJ"
                    >
                      New Jersey
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="NM"
                    >
                      New Mexico
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="NY"
                    >
                      New York
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="NC"
                    >
                      North Carolina
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="ND"
                    >
                      North Dakota
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="MP"
                    >
                      Northern Mariana Islands
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="OH"
                    >
                      Ohio
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="OK"
                    >
                      Oklahoma
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="OR"
                    >
                      Oregon
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="PW"
                    >
                      Palau
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="PA"
                    >
                      Pennsylvania
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="PR"
                    >
                      Puerto Rico
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="RI"
                    >
                      Rhode Island
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="SC"
                    >
                      South Carolina
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="SD"
                    >
                      South Dakota
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="TN"
                    >
                      Tennessee
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="TX"
                    >
                      Texas
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="UT"
                    >
                      Utah
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="VT"
                    >
                      Vermont
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="VI"
                    >
                      Virgin Islands
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="VA"
                    >
                      Virginia
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="WA"
                    >
                      Washington
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="WV"
                    >
                      West Virginia
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="WI"
                    >
                      Wisconsin
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="WY"
                    >
                      Wyoming
                    </option>
                  </select>
                </span>
                <input
                  class=""
                  icon="caret-down"
                  name="stateProvinceRegion"
                  placeholder="State"
                />
                <i
                  class="icon icon_caret-down"
                />
              </div>
            </div>
            <div
              class="field phone-number-field phone-number-field--international"
            >
              <div
                class="labeled input large"
              >
                <div
                  class="input--label mr4"
                >
                  +1
                </div>
                <input
                  label="+1"
                  name="phoneNumber"
                  placeholder="SHARED__PLACEHOLDER__PHONE_NUMBER"
                  type="tel"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="parent-or-guardian-form--disclaimer-text"
        >
          <span>
            I confirm the accuracy of this contact information and the person is someone other than their purchaser. Southwest Airlines® may contact them.

I acknowledge and confirm the
          </span>
           
          <a
            class="pblue"
            href="#"
          >
            Young Traveler parent consent
          </a>
        </div>
      </div>
      <div
        class="parent-or-guardian-form--continue-button"
      >
        <button
          class="button button--fluid larger button--yellow"
          type="submit"
        >
          AIR_BOOKING__PARENT_OR_GUARDIAN_FORM__CONTINUE
        </button>
      </div>
    </fieldset>
  </form>
</div>
`;

exports[`ParentOrGuardianForm should render the component correctly when isEditMode is true 1`] = `
<div>
  <form
    class="form"
    name="AIR_BOOKING_PARENT_OR_GUARDIAN_FORM"
    novalidate=""
  >
    <fieldset>
      <div>
        <div
          class="page-header action-bar"
          id=""
        >
          <ul
            class="action-bar-buttons action-bar--left-buttons"
          />
          <div
            class="action-bar--container"
          >
            <div
              class="action-bar--title"
            >
              <span
                class="action-bar--main-title"
              >
                AIR_BOOKING__YOUNG_TRAVELER_EDIT__PAGE_TITLE
              </span>
              <span
                class="action-bar--sub-title"
              />
            </div>
          </div>
          <ul
            class="action-bar-buttons action-bar--right-buttons"
          >
            <li
              class="action-bar-buttons--item"
            >
              <button
                class="button"
                type="submit"
              >
                SHARED__BUTTON_TEXT__DONE
              </button>
            </li>
          </ul>
        </div>
        <div />
      </div>
      <div
        class="parent-or-guardian-form--container"
      >
        <p
          class="parent-or-guardian-form--info-text"
        >
          You must complete the form entirely including a phone number your parent/guardian contact can be reached in case of emergency. Once you have completed this purchase, any changes to the parent/guardian information must be completed at the airport.
        </p>
        <div
          class="fields pt5 grouped"
        >
          <label
            class="fields--label"
          >
            AIR_BOOKING__PARENT_OR_GUARDIAN_FORM__PARENT_OR_GUARDIAN_LABEL
          </label>
          <div
            class="field"
          >
            <div
              class="input"
            >
              <input
                class=""
                maxlength="30"
                name="firstName"
                placeholder="SHARED__PLACEHOLDER__FIRST_NAME"
                value=""
              />
            </div>
          </div>
          <div
            class="field"
          >
            <div
              class="input"
            >
              <input
                class=""
                maxlength="30"
                name="lastName"
                placeholder="SHARED__PLACEHOLDER__LAST_NAME"
                value=""
              />
            </div>
          </div>
          <div
            class="field"
          >
            <div
              class="input"
            >
              <input
                class=""
                maxlength="20"
                name="relationship"
                placeholder="SHARED__PLACEHOLDER__RELATIONSHIP"
                value=""
              />
            </div>
          </div>
        </div>
        <div>
          <div
            class="fields mt5 grouped"
          >
            <label
              class="fields--label"
            >
              AIR_BOOKING__PARENT_OR_GUARDIAN_FORM__CONTACT_INFORMATION_LABEL
            </label>
            <div
              class="field"
            >
              <div
                class="country-code-nav-item-field"
                data-qa="country-code-nav-item-field"
              >
                <a
                  class="nav-item-link input"
                >
                  <div
                    class="country-code-nav-item-field--value"
                    data-qa="country-code-nav-item-field-value"
                  >
                    United States of America
                     - 
                    US
                  </div>
                  <i
                    class="nav-item-link--icon icon icon_keyboard-arrow-right"
                  />
                </a>
              </div>
            </div>
            <div
              class="field"
            >
              <div
                class="input large"
              >
                <input
                  class=""
                  maxlength="40"
                  name="addressLine1"
                  placeholder="Street address"
                  value=""
                />
              </div>
            </div>
            <div
              class="field"
            >
              <div
                class="input large"
              >
                <input
                  class=""
                  maxlength="40"
                  name="addressLine2"
                  placeholder="Street address 2 (optional)"
                  value=""
                />
              </div>
            </div>
            <div
              class="field"
            >
              <div
                class="input large"
              >
                <input
                  name="zipOrPostalCode"
                  placeholder="ZIP code"
                  type="tel"
                  value=""
                />
              </div>
            </div>
            <div
              class="field"
            >
              <div
                class="input large"
              >
                <input
                  class=""
                  name="city"
                  placeholder="City"
                  value=""
                />
              </div>
            </div>
            <div
              class="field"
            >
              <div
                class="province-field_us select-input icon-fixed formatted input large"
              >
                <span
                  class="formatted-input formatted-input--fill"
                >
                  <select
                    class="dropdown empty large"
                    data-qa="select-dropdown"
                    name="stateProvinceRegion"
                    placeholder="State"
                  >
                    <option
                      class="placeholder"
                      data-qa="dqa-enabled-option"
                      value=""
                    >
                      State
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="AL"
                    >
                      Alabama
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="AK"
                    >
                      Alaska
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="AS"
                    >
                      American Samoa
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="AZ"
                    >
                      Arizona
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="AR"
                    >
                      Arkansas
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="AA"
                    >
                      Armed Forces Americas (except Canada)
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="AE"
                    >
                      Armed Forces
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="AP"
                    >
                      Armed Forces Pacific
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="CA"
                    >
                      California
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="CO"
                    >
                      Colorado
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="CT"
                    >
                      Connecticut
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="DE"
                    >
                      Delaware
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="DC"
                    >
                      District of Columbia
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="FM"
                    >
                      Federated States of Micronesia
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="FL"
                    >
                      Florida
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="GA"
                    >
                      Georgia
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="GU"
                    >
                      Guam
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="HI"
                    >
                      Hawaii
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="ID"
                    >
                      Idaho
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="IL"
                    >
                      Illinois
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="IN"
                    >
                      Indiana
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="IA"
                    >
                      Iowa
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="KS"
                    >
                      Kansas
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="KY"
                    >
                      Kentucky
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="LA"
                    >
                      Louisiana
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="ME"
                    >
                      Maine
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="MH"
                    >
                      Marshall Islands
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="MD"
                    >
                      Maryland
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="MA"
                    >
                      Massachusetts
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="MI"
                    >
                      Michigan
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="MN"
                    >
                      Minnesota
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="MS"
                    >
                      Mississippi
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="MO"
                    >
                      Missouri
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="MT"
                    >
                      Montana
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="NE"
                    >
                      Nebraska
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="NV"
                    >
                      Nevada
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="NH"
                    >
                      New Hampshire
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="NJ"
                    >
                      New Jersey
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="NM"
                    >
                      New Mexico
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="NY"
                    >
                      New York
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="NC"
                    >
                      North Carolina
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="ND"
                    >
                      North Dakota
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="MP"
                    >
                      Northern Mariana Islands
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="OH"
                    >
                      Ohio
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="OK"
                    >
                      Oklahoma
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="OR"
                    >
                      Oregon
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="PW"
                    >
                      Palau
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="PA"
                    >
                      Pennsylvania
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="PR"
                    >
                      Puerto Rico
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="RI"
                    >
                      Rhode Island
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="SC"
                    >
                      South Carolina
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="SD"
                    >
                      South Dakota
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="TN"
                    >
                      Tennessee
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="TX"
                    >
                      Texas
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="UT"
                    >
                      Utah
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="VT"
                    >
                      Vermont
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="VI"
                    >
                      Virgin Islands
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="VA"
                    >
                      Virginia
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="WA"
                    >
                      Washington
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="WV"
                    >
                      West Virginia
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="WI"
                    >
                      Wisconsin
                    </option>
                    <option
                      data-qa="dqa-enabled-option"
                      value="WY"
                    >
                      Wyoming
                    </option>
                  </select>
                </span>
                <input
                  class=""
                  icon="caret-down"
                  name="stateProvinceRegion"
                  placeholder="State"
                />
                <i
                  class="icon icon_caret-down"
                />
              </div>
            </div>
            <div
              class="field phone-number-field phone-number-field--international"
            >
              <div
                class="labeled input large"
              >
                <div
                  class="input--label mr4"
                >
                  +1
                </div>
                <input
                  label="+1"
                  name="phoneNumber"
                  placeholder="SHARED__PLACEHOLDER__PHONE_NUMBER"
                  type="tel"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="parent-or-guardian-form--disclaimer-text"
        >
          <span>
            I confirm the accuracy of this contact information and the person is someone other than their purchaser. Southwest Airlines® may contact them.

I acknowledge and confirm the
          </span>
           
          <a
            class="pblue"
            href="#"
          >
            Young Traveler parent consent
          </a>
        </div>
      </div>
    </fieldset>
  </form>
</div>
`;
